strawberry/utils/typing.py

```diff
@@ -249,6 +255,36 @@ def eval_type(
             assert ast_unparse
             type_ = ForwardRef(ast_unparse(parsed))
 
+        # When using forward refs, the whole
+        # Annotated[SomeType, strabwerry.lazy("type.module")] is a forward ref,
+        # and trying to _eval_type on it will fail. Take a different approach
+        # here to resolve lazy types by execing the annotated args and resolving
+        # the type directly.
+        annotated_match = _annotated_re.match(type_.__forward_arg__)
+        if annotated_match:
+            gdict = annotated_match.groupdict()
+            ldict = (localns or {}).copy()
+            # Exec the remaining annotated args to get their real values and
+            # put the result in ldict["args"]
+            exec(f'args = ({gdict["args"]}, )', globalns, ldict)
```

Comment:
could we use eval here?

btw: this makes me a bit antsy :D

Reply:
Sure!Totally forgot about `eval` lol

> btw: this makes me a bit antsy :D

hahaha me too. The major issue here is that the whole thing is a string, but we need its value to resolve the lazy part. Don't know if there's a better option for this, it was the best I could think of =P (but open to suggestions)

Reply:
we should check how pydantic deals with these things, but we can do that another time 😊

Reply:
I did a quick check, and they have a similar solution to us: https://github.com/pydantic/pydantic/blob/c600c010418c46cec1f3a0fe8fb6ca4be4a7949f/pydantic/_internal/_typing_extra.py#L319

I don't think they workaround this specific issue (maybe they don't have it?). They call [ForwardRef._evaluate](https://github.com/python/cpython/blob/main/Lib/typing.py#L862) directly, but we can't do that because we don't have the module in globals/locals, and the only way to retrieve it is to execute the `strawberry.lazy(...)` argument (or parse it and call `__import__` there).

We might want to revisit how lazy works another time to allow us to better support this :)